### PR TITLE
Basic Coordinates class added

### PR DESCRIPTION
**Pull request by Rogal255**

### CODE REVIEW

File: include/game/Coordinates.hpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            : value(coordX) {}
+
+        constexpr bool operator==(const CoordX& other) const {
+            return value == other.value;
+        }
+
+        int value{0};
+    };
+
+    struct CoordY {
```
Comment (runer53248) [inline include/game/Coordinates.hpp:18]: Duplication.

File: include/game/Coordinates.hpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+public:
+    struct CoordX {
+        explicit constexpr CoordX(int coordX)
+            : value(coordX) {}
+
+        constexpr bool operator==(const CoordX& other) const {
+            return value == other.value;
+        }
+
+        int value{0};
```
Comment (runer53248) [inline include/game/Coordinates.hpp:15]: Can be const. Dont need to set to 0.

File: test/CoordinatesTest.cpp [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#include <gtest/gtest.h>
+#include <utility>
+#include "game/Coordinates.hpp"
+
+TEST(CoordinatesTest, Basic) {
+    constexpr int x{10};
+    constexpr int y{-5};
```
Comment (runer53248) [inline test/CoordinatesTest.cpp:7]: Create  needed type directly.

File: include/game/Coordinates.hpp [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    };
+
+    constexpr Coordinates(const CoordX& posX, const CoordY& posY)
+        : posX_(posX), posY_(posY) {}
+
+    constexpr std::pair<CoordX, CoordY> getPosition() const {
```
Comment (runer53248) [inline include/game/Coordinates.hpp:32]: Not sure we need this method (Coordinates is position). Maybe distance or some kind of lerp will be enough.
Assuming that we will have separate types for coordinates, consider writing the operator == for each individual and for Coordinates as whole.

**Review by runer53248 — CHANGES_REQUESTED** [review]

File: include/game/Coordinates.hpp (hunk 1 above)
Comment (runer53248) [reply include/game/Coordinates.hpp:18]: Consider template for axis to eliminate duplication:
    Coordinates(Coordinate<Axis::X> posX, Coordinate<Axis::Y> posY)

Comment (Rogal255) [reply include/game/Coordinates.hpp:18]: I'm not sure how to implement that.

Comment (Rogal255) [reply include/game/Coordinates.hpp]: Refactored

File: include/game/Coordinates.hpp [hunk 5]
```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <utility>
```
Comment (runer53248) [inline include/game/Coordinates.hpp]: Not needed.

File: test/CoordinatesTest.cpp [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#include <gtest/gtest.h>
+#include "game/Coordinates.hpp"
+
+TEST(CoordinatesTest, Basic) {
+    constexpr Coordinates first((Coordinates::Coord<Axis::X>(10)), (Coordinates::Coord<Axis::Y>(20)));
+    constexpr Coordinates second((Coordinates::Coord<Axis::X>(10)), (Coordinates::Coord<Axis::Y>(20)));
+    ASSERT_EQ(first, second);
+}
```
Comment (runer53248) [inline test/CoordinatesTest.cpp:8]: Add test for different coords.

**Review by runer53248 — CHANGES_REQUESTED** [review]

**Review by runer53248 — APPROVED** [review]

**Pull request merged**